tests/migrations/test_writer.py

```diff
@@ -650,3 +654,11 @@ def deconstruct(self):
 
         string = MigrationWriter.serialize(models.CharField(default=DeconstructibleInstances))[0]
         self.assertEqual(string, "models.CharField(default=migrations.test_writer.DeconstructibleInstances)")
+
+    def test_register(self):
+        with self.assertRaisesMessage(ValueError, 'Cannot serialize: (1+2j)'):
+            self.assertSerializedEqual(complex(1, 2))
+        with self.assertRaisesMessage(ValueError, "'TestModel1' must inherit from 'BaseSerializer'."):
+            MigrationWriter.register_serializer(complex, TestModel1)
+        MigrationWriter.register_serializer(complex, ComplexSerializer)
```

Comment:
The serializer must be unregistered as well to avoid polluting the global state.

Reply:
Add an `unregister_serializer()` method and call it in the test.

Reply:
thanks i added it